java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -727,6 +727,75 @@ class DatabaseImpl(
         }
     }
 
+    override suspend fun removeExpiredEntities() {
+        // Fix the time before starting the transaction.
+        val nowMillis = JvmTime.currentTimeMillis
+        writableDatabase.transaction {
+            // Find all expired entities.
+            val storageKeyIds = rawQuery(
+                "SELECT storage_key_id FROM entities WHERE expiration_timestamp < ?",
+                arrayOf(nowMillis.toString())
+            ).map { it.getLong(0) }.toSet()
+
+            // Remove field values.
+            delete(
+                TABLE_FIELD_VALUES,
+                "entity_storage_key_id in (?)",
+                arrayOf(storageKeyIds.joinToString())
+            )
+
+            // TODO: remove from collection_entries all references to these entities.
+
+            // Clean up unused values as they can contain sensitive data.
+            // This query will return all field value ids being referenced by collection or 
+            // singleton fields.
+            val usedFieldIdsQuery =
+                """
+                    (SELECT
```

Comment:
Minor nit: I'd recommend removing the parens here and using them where it's used below. It's more obviously correct, is all.